File: AlogoExpert-Done-Q/0609removeDuplicate.ts
export class LinkedList {
  value: number;
  next: LinkedList | null;

  constructor(value: number) {
    this.value = value;
    this.next = null;
  }
}
export function removeDuplicatesFromLinkedList(linkedList: LinkedList) {
  let current: LinkedList | null = linkedList;
  let next: LinkedList | null = linkedList.next;

  while (!!current) {
    if (current.value !== next?.value) {
      current.next = next;
      current = next;
    }
    next = next?.next || null;
  }
  return linkedList;
}

File: AlogoExpert-OthersLang/0609removeDuplicate.py
# O(n) time | O(1) space
# because we all have to do is look trough all of the nodes in our linked list. We only have to look at wach of them once. And we don't use any external space either.
# Other requirement is the list in ascending order always.
class LinkedList:
    def __init__(self, value):
        self.value = value
        self.next = None

def removeDuplicatesFromLinkedList(linkedList):
    currentNode = linkedList
    while currentNode is not None:
        nextDistinctNode = currentNode.next
        while nextDistinctNode is not None and nextDistinctNode.value == currentNode.value:
            nextDistinctNode = nextDistinctNode.next
        
        currentNode.next = nextDistinctNode
        currentNode = nextDistinctNode
    return linkedList